packages/server/src/server.js

```diff
@@ -183,62 +184,43 @@ export default class Server {
     }
   }
 
-  listen(port, host, socket) {
-    return new Promise((resolve, reject) => {
-      if (!socket && typeof this.options.server.socket === 'string') {
-        socket = this.options.server.socket
-      }
-
-      const args = { exclusive: false }
-
-      if (socket) {
-        args.path = socket
-      } else {
-        args.port = port || this.options.server.port
-        args.host = host || this.options.server.host
+  normalizeURL(options, server) {
+    const address = server.address()
+    if (options.protocol !== 'socket') {
+      switch (address.address) {
```

Comment:
Here's another pattern I like to use for this:

```
options.host = {
  '127.0.0.1'': 'localhost',
  '0.0.0.0: ip.address()
}[address.address] || options.host
```

Reply:
I usually prefer a lookup object or a map too (instead of switch/case or if cascades)

Reply:
Yep, it looks graceful, but `ip.address` will be executed even host is not `0.0.0.0`. 😄

Reply:
@clarkdo good point :)

Reply:
```js
options.host = ({
  '127.0.0.1': () => 'localhost',
  '0.0.0.0': () => ip.address()
}[address.address] || () => options.host)()
```

Suddenly not so graceful anymore 😄